src/lib/server/function_words.js

```diff
@@ -0,0 +1,88 @@
+/** @type {[string, string][]} */
+const FUNCTION_WORDS = [
+	['begin', 'Sets the Aspect feature of a Verb to Inceptive.'],
+	['certainly', 'Sets the Polarity feature of a Verb to Emphatic Affirmative/Negative.'],
+	['continue', 'Sets the Aspect feature of a Verb to Continuative.'],
+	['could', 'Used with so-A. Otherwise, use "be able".'],
+	['definitely', 'Sets the Mood feature of a Verb to Definite Potential.'],
+	['extremely', 'Sets the Degree feature of an Adjective to Extremely Intensified.'],
+	['finish', 'Sets the Aspect feature of a Verb to Completive.'],
+	['from', 'Usually indicates the Source argument of a Verb.'],
+	['here', "Becomes 'at this place'."],
+	['least', "Sets the Degree feature of an Adjective to 'least'."],
+	['may', 'Sets the Mood feature of a Verb to Permissive May.'],
+	['might', 'Sets the Mood feature of a Verb to Might Potential, or used with if-A.'],
+	['must', 'Sets the Mood feature of a Verb to Must Obligation.'],
+	['probably', 'Sets the Mood feature of a Verb to Definite Potential.'],
+	['should', 'Sets the Mood feature of a Verb to Should Obligation.'],
+	['start', 'Sets the Aspect feature of a Verb to Inceptive.'],
+	['stop', 'Sets the Aspect feature of a Verb to Cessative.'],
+	['there', "Becomes 'at this place', except when used with be-E."],
+	['too', "Sets the Degree feature of an Adjective to 'too'."],
+	['very', 'Sets the Degree feature of an Adjective to Intensified.'],
+	['with', 'Sometimes used to indicate the Instrument argument of a Verb.'],
+	['would', 'Used with so-C, if-B, or if-C. Otherwise, cannot be used.'],
+	['which', 'Indicates an interrogative Noun or a relative clause.'],
+]
+
+/**
+ * @param {ConceptSearchFilter} filter
+ * @returns {Concept[]}
+ */
+export function get_function_words(filter) {
+	if (!['all', ''].includes(filter.category) || filter.scope === 'glosses') {
+		return []
+	}
+
+	const QUERY_REGEX = /^(?<pre_wildcard>[*%#]?)(?<term>.+?)(?<post_wildcard>[*%#]?)$/
+	const match = filter.q.match(QUERY_REGEX)
+	if (!match?.groups) {
+		return []
+	}
+
+	const { pre_wildcard, term, post_wildcard } = match.groups
+	return FUNCTION_WORDS.filter(get_word_filter(pre_wildcard, term, post_wildcard)).map(transform_function_word)
+}
+
+/**
+ * @param {string} pre_wildcard
+ * @param {string} term
+ * @param {string} post_wildcard
+ * @returns {(word_and_gloss: [string, string]) => boolean}
+ */
+function get_word_filter(pre_wildcard, term, post_wildcard) {
+	const lower_term = term.toLowerCase()
+	
+	if (!pre_wildcard && post_wildcard) {
```

Comment:
what are your thoughts for something like this instead of the if/else approach?  (no pressure, just curious)

```js
const filterMap = {
		'00': ([word]) => word === lower_term,
		'10': ([word]) => word.endsWith(lower_term),
		'01': ([word]) => word.startsWith(lower_term),
		'11': ([word]) => word.includes(lower_term),
	};

	const key = `${pre_wildcard ? 1 : 0}${post_wildcard ? 1 : 0}`;
	return filterMap[key];
```

Reply:
😁 the only reason I like it better is because it reads like a truth table to me 🤓 (and of course it eliminates the `if`'s)